proto/redpanda/runtime/v1alpha1/input.proto

```diff
@@ -0,0 +1,125 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package redpanda.runtime.v1alpha1;
+
+option go_package = "github.com/redpanda-data/connect/v4/internal/dynamic/plugin/runtimepb";
+
+import "redpanda/runtime/v1alpha1/message.proto";
+
+
+// BatchInput is an interface implemented by Benthos inputs that produce messages
+// in batches, where there is a desire to process and send the batch as a logical
+// group rather than as individual messages.
+// 
+// Calls to ReadBatch should block until either a message batch is ready to process,
+// the connection is lost, or the provided context is cancelled.
+service BatchInputService {
```

Comment:
In almost all doc strings we have the note about context cancelation "or the provided context is cancelled" is it relevant in the context of gRPC?

Reply:
Switched to talking about RPC timeouts